### PR TITLE
Review why Julia

### PR DESCRIPTION
**Pull request by rikhuijzer**
I've tried to keep the PR a bit short to avoid huge discussions. Let me know if you have doubts about one of my changes!

### CODE REVIEW

File: contents/why_julia.md [hunk 1]
```diff
@@ -392,7 +401,7 @@ If you would like to have a deep dive on how Julia is being used, we have some *
 
 1. NASA uses Julia in a supercomputer to analyze the ["Largest Batch of Earth-Sized Planets Ever Found"](https://exoplanets.nasa.gov/news/1669/seven-rocky-trappist-1-planets-may-be-made-of-similar-stuff/) and achieve a whopping **1,000x speedup** to catalog 188 million astronomical objects in 15 minutes.
 2. [The Climate Modeling Alliance (CliMa)](https://clima.caltech.edu/) is using mostly Julia to **model climate in the GPU and CPU**.
-Launched in 2018 in collaboration with researchers at Caltech, the NASA Jet Propulsion Laboratory, and the Naval Postgraduate School, CliMA is utilizing recent progress in computational science to develop an Earth system model that can predict droughts, heat waves, and rainfall with unprecedented precision.
```
Comment (rikhuijzer) [inline contents/why_julia.md:395]: The Oxford comma is only for lists of three or more items.

File: contents/why_julia.md [hunk 2]
```diff
@@ -133,8 +136,20 @@ We don't know any other serious language that would want to be hard and slow, so
...
+For example, if you define a function taking one argument and pass an integer into the function, then Julia will create a _specialized_ `MethodInstance`.
+The next time that you pass an integer to the function, Julia will lookup the `MethodInstance` that was created earlier and refer execution to that.
+Now, the **great** trick is that you can also do this inside a function that calls a function.
+For example, if some data type is passed into function `f` and `f` calls function `g` and the data types passed to `g` are known and always the same, then the generated function `g` can be hardcoded into function `f`!
+This means that Julia doesn't even have to lookup `MethodInstances` anymore, and the code can run very efficiently.
+The tradeoff, here, is that there are cases where earlier assumptions about the hardcoded `MethodInstances` are invalidated.
+Then, the `MethodInstance` has to be recreated which takes time.
+Also, the tradeoff is that it takes time to infer what can be hardcoded and what not.
+This explains why it can often take very long before Julia does the first thing:
+in the background, it is optimizing your code.
```
Comment (rikhuijzer) [inline contents/why_julia.md:151]: I guess that you maybe get this explanation, but people new to Julia do not. Do you have tips on how I could make this explanation more clear, @storopoli?

File: contents/why_julia.md (hunk 1 above)
Comment (storopoli) [reply contents/why_julia.md:395]: What is oxford comma? 😂

File: contents/why_julia.md (hunk 2 above)
Comment (storopoli) [reply contents/why_julia.md:151]: People don't know what LLVM code is. I would prefer to "dummyfy" a little bit and just say machine code. Or we could add something like

> very efficient LLVM code which translates to machine code.

Comment (rikhuijzer) [reply contents/why_julia.md:153]: Okay. I've added an explanation. Is it clear enough now, you think?

Comment (storopoli) [reply contents/why_julia.md:153]: Yes!

**Review by storopoli — APPROVED** [review]
Great addition @rikhuijzer. Thanks!

**Comment by rikhuijzer** [comment]
Thanks for the review again :heart:

**Pull request merged**